training/model_management/src/azureml/model/mgmt/downloader/downloader.py

```diff
@@ -44,8 +43,19 @@ def __init__(self, model_id: str):
         """
         self._model_id = model_id
         self._model_uri = self.HF_ENDPOINT + f"/{model_id}"
-        self._model_info = None
         self._hf_api = HfApi(endpoint=self.HF_ENDPOINT)
+        self._is_valid_id = False
+        self._model_info = None
+
+    @property
+    def is_valid_id(self):
+        """Returns true if Hugging face model id is valid."""
+        model_list = [
+            model.modelId for model in self._hf_api.list_models(filter=ModelFilter(model_name=self._model_id))
```

Comment:
This looks like an expensive operation. You previously had caching, but now it will make the call every time.